docs/migrating-to-0.45.0/README.md

```diff
@@ -0,0 +1,191 @@
+# Migrating to v0.45.0
+
+Starship v0.45.0 is a release containing breaking changes, in preparation for the big v1.0.0. We have made some major changes around how configuration is done on the prompt, to allow for a greater degree of customization.
+
+This guide is intended to walk you through the breaking changes.
+
+## `prompt_order` has been replaced by a root-level `format`
+
+Previously to v0.45.0, `prompt_order` would accept an array of module names
+in the order which they should be rendered by Starship.
+
+Starship v0.45.0 will instead be accepting a `format` value, allowing for customization of the prompt outside of the modules themselves.
+
+**Example pre-v0.45.0 configuration**
+
+```toml
+prompt_order = [
+  "username",
+  "hostname",
+  "directory",
+  "git_branch",
+  "git_commit",
+  "git_state",
+  "git_status",
+  "cmd_duration",
+  "custom",
+  "line_break",
+  "jobs",
+  "battery",
+  "time",
+  "character",
+]
+```
+
+**Example v0.45.0 configuration**
+
+```toml
+format = """\
+  $username\
+  $hostname\
+  $directory\
+  $git_branch\
+  $git_commit\
+  $git_state\
+  $git_status\
+  $cmd_duration\
+  $custom\
+  $line_break\
+  $jobs\
+  $battery\
+  $time\
+  $character\
+  """
+```
+
+## Module `prefix` and `suffix` will be replaced by `format`
+
+Previously to v0.45.0, some modules would accept `prefix` and/or `suffix`
+in order to stylize the way that modules are rendered.
+
+Starship v0.45.0 will instead be accepting a `format` value, allowing for further
+customization of how modules are rendered. Instead of defining a prefix and suffix
+for the context-based variables, the variables can now be substituted from within
+a format string, which represents the module's output.
+
+**Example pre-v0.45.0 configuration**
+
+```toml
+[cmd_duration]
+prefix = "took "
+```
+
+**Example v0.45.0 configuration**
+
+```toml
+[cmd_duration]
+# $duration – The command duration (e.g. "15s")
+# $style    – The default style of the module (e.g. "bold yellow")
+format = "took [$duration]($style)"
+```
+
+### Affected Modules
+
```

Comment:
Character also has breaking changes

```suggestion

#### Character

| Deprecated Property | Replacement |
| ------------------- | ----------- |
| `symbol` | `success_symbol` |
| `use_symbol_for_status` | `success_symbol`, `error_symbol` |
| `style_success` | `success_symbol` |
| `style_failure` | `error_symbol` |

```diff
[character]
-- symbol = "➜"
-- error_symbol = "✗"
-- use_symbol_for_status = true
++ success_symbol = "[➜](bold green) "                                                                                     
++ error_symbol = "[✗](bold red) "
```

```

Reply:
I'm pretty sure that the new character configuration is the following:

```toml
success_symbol = "[❯](bold green)"
error_symbol = "[❯](bold red)"
```

Reply:
Ah yep you are right, I just copied my local config.